lib/remap-functions/src/encode_base64.rs

```diff
@@ -0,0 +1,143 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeBase64;
+
+impl Function for EncodeBase64 {
+    fn identifier(&self) -> &'static str {
+        "encode_base64"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                accepts: |v| matches!(v, Value::Bytes(_)),
+                required: true,
+            },
+            Parameter {
+                keyword: "padding",
+                accepts: |v| matches!(v, Value::Boolean(_)),
+                required: false,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+        let padding = arguments.optional("padding").map(Expr::boxed);
+
+        Ok(Box::new(EncodeBase64Fn { value, padding }))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct EncodeBase64Fn {
+    value: Box<dyn Expression>,
+    padding: Option<Box<dyn Expression>>,
+}
+
+impl Expression for EncodeBase64Fn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let value = self.value.execute(state, object)?.try_bytes()?;
+
+        let padding = match &self.padding {
+            Some(p) => Some(p.execute(state, object)?.try_boolean()?),
+            None => None,
+        };
+
+        let config = match padding {
```

Comment:
With the above change, this could simply be:

```
let config = match padding {
  true => base64::STANDARD,
  false => base64::STANDARD_NO_PAD,
}
```

Reply:
I don't have a rock-solid citation but I believe that matching on a Boolean is considered non-idiomatic and that if/else should be used instead. There is, for example, a Clippy lint for that: https://rust-lang.github.io/rust-clippy/master/#match_bool.

Reply:
Aha, I didn't realize there was a clippy lint for that. I'm content with if/else too:

```
let config = if padding {
  base64::STANDARD
} else {
  fbase64::STANDARD_NO_PAD,
}
```